src/sage/combinat/words/morphism.py

```diff
@@ -1725,11 +1728,19 @@ def is_uniform(self, k=None):
             False
             sage: tau.is_uniform(k=2)
             True
+
+        TESTS::
+
+            sage: phi = WordMorphism('')
+            sage: phi.is_uniform()
+            True
         """
         if k is None:
-            return len(set(w.length() for w in self.images())) == 1
-        else:
-            return all(w.length() == k for w in self.images())
+            try:
+                k = self.images()[0].length()
+            except IndexError:
+                return True
+        return all(w.length() == k for w in self.images())
```

Comment:
How about rewriting it using an iterator? Just bikeshedding, feel free to ignore.
```
images = iter(self.images())
if k is None:
    try:
        k = next(images).length()
    except StopIteration:
        return True
return all(w.length() == k for w in images)

Reply:
well, this is a list, so there is no point, I think

Reply:
Fine